phinet/networks.py

```diff
@@ -6,14 +6,90 @@
     get_xpansion_factor,
 )
 from .blocks import PhiNetConvBlock
+import phinet
 
 from pathlib import Path
 from torchinfo import summary
 import torch.nn as nn
 import torch
+from huggingface_hub import hf_hub_download
+from huggingface_hub.utils import EntryNotFoundError
+from types import SimpleNamespace
 
 
 class PhiNet(nn.Module):
+    @classmethod
+    def from_pretrained(
+        cls,
+        dataset,
+        alpha,
+        beta,
+        t_zero,
+        num_layers,
+        num_classes,
+        resolution,
+        device=None,
+    ):
+        def num_class_error():
+            raise ValueError(
+                "Can't load model because num_classes does not match with dataset."
+            )
+
+        repo_dir = f"mbeltrami/{dataset}"
+        file_to_choose = f"\
+phinet_a{float(alpha)}_b{float(beta)}_tzero{float(t_zero)}_Nlayers{num_layers}\
+_res{resolution}{phinet.datasets_info[dataset]['ext']}\
+"
+
+        assert (
+            num_classes == phinet.datasets_info[dataset]["Nclasses"]
+        ), num_class_error()
```

Comment:
For this, we can directly put a string; it should raise AssertionError by default -- I forgot about this.
Something like:
```
assert (
    num_classes == phinet.datasets_info[dataset]["Nclasses"]
),  "messaggio di errore"
```

-- see [here](https://www.w3schools.com/python/ref_keyword_assert.asp#:~:text=Definition%20and%20Usage,False%2C%20check%20the%20example%20below.) alse